src/v/metrics/metrics_registry.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/metrics.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+/*
+ * Class which allows tracking all our registered metrics and their
+ * aggregation labels for both values of our aggregate_metrics config flag.
+ *
+ * Hence, it can be used to easily swap the aggregation labels for all
+ * registered metrics when the config value changes. See
+ * `aggregate_metrics_watcher`
+ *
+ * We use it as a global threadlocal instance so that it can easily be accessed
+ * from all probes.
+ */
```

Comment:
Somewhere in a comment we should be clear what "aggregation labels" are, because at least I find it a bit confusing (i.e., I might guess the opposite). 

"Aggregation labels" are labels that are _dropped_ turning two metrics which different only in "aggregation labels" into a single one (with only non-aggregation labels renaming) after summing their values, right?

Reply:
Have added it here as suggested.